client/src/hooks/permissions/index.ts

```diff
@@ -0,0 +1,25 @@
+import { useProfile } from 'hooks/profile';
+
+import type { Permission, RoleName } from './enums';
+
+export function usePermissions() {
+  const { data, isLoading } = useProfile();
+
+  const roles: RoleName[] = [];
+  const permissions: Permission[] = [];
+
+  data?.roles?.forEach((role) => {
+    roles.push(role.name);
+    role.permissions?.forEach((permission) => permissions.push(permission));
+  });
+
+  const hasRole = (role: RoleName) => {
+    return !isLoading && !!roles?.includes(role);
+  };
+
+  const hasPermission = (permission: Permission) => {
+    return !isLoading && !!permissions?.includes(permission);
+  };
```

Comment:
I think this wouldn't work, as you would be passing a enum value which would be transpiled to a string, and this is how [the permission entity looks](https://api.dev.landgriffon.com/swagger/#/User/UsersController_userMetadata) like.

Unless you specifically want the backend to retrieve permissions as an array of strings, happy to discuss that. But by default we will nest entities by its original shape.

Also maybe the naming on our side it's not the best? happy to discuss that as well

Reply:
We know it couldn't work, it's like that because it's easier to work with a array of string and we can parse what we get from API. And last reason, because it's not documented the format of the permissions in Swagger and we thought it was still missing.

Reply:
All clear sir